c7n/resources/account.py

```diff
@@ -1839,3 +1839,45 @@ def process(self, resources, event=None):
         if state == bool(sechub):
             return resources
         return []
+
+
+@filters.register('lakeformation-s3-cross-account')
+class LakeformationFilter(Filter):
+    """To check if S3 bucket i.e. registered for Lakeformation belongs to the same account or not.
+
+    :example:
+
+    .. code-block:: yaml
+
+       policies:
+         - name: lakeformation-cross-account-bucket
+           resource: aws.account
+           filters:
+            - type: lakeformation-s3-cross-account
+
+    """
+
+    def process(self, resources, event=None):
+        if not resources[0].get('c7n:lakeformations3crossaccount'):
+            Buckets = self.manager.get_resource_manager('s3').resources()
+            resourcelist = []
+            client = local_session(self.manager.session_factory).client('lakeformation')
+            resourcelist = client.list_resources()
+            filtered_resources_set = set()
+            filtered_resources = []
+            for resource in resourcelist['ResourceInfoList']:
+                arnName = resource['ResourceArn']
+                arnName = arnName.partition('arn:aws:s3:::')[2]
+                isMatch = 0
+                for bucket in Buckets:
```

Comment:
I think we can simplify it further by not using nested for loops

Reply:
Done. Please check.